magento/magento2
# improve Magento\Catalog\Model\ImageUploader error handler

### Description (*)
Magento\Catalog\Model\ImageUploader is catching detailed exceptions and turning it into generic "Something went wrong while saving the file(s)." Data on what actually went wrong is gone.
This fix logs exception and passes previous exception to constructor not to loose this info.

Also I would like to discuss if there are some guidelines on handling such cases in Magento 2.
In my humble opinion in development mode we show detailed info and in production mode we show "something went wrong" and log exception so catching exception just to show "Something went wrong while saving the file" in both modes and loose debug info does not make much sense and I would remove both this try/catches at all.

### Manual testing scenarios (*)

1) change media dir permissions to not writable
2) upload a file in admin panel
3) see info on what folder lacks permissions in error log.

### Contribution checklist (*)
